src/frontend/src/session.rs

```diff
@@ -668,6 +669,30 @@ impl SessionImpl {
         Ok(connection.clone())
     }
 
+    pub fn get_source_by_name(
```

Comment:
It seems a little strange to me that we implement a specialized catalog lookup function directly on `SessionImpl`. cc @yezizp2012 Do you have better ideas?

Reply:
Yes, it's quite strange to do so and it seems like we already introduced several specialized catalog functions in `SessionImpl`. How about moving them into `CatalogReader` and let the caller access the reader and call those functions directly?

Reply:
@StrikeW What do you think? 🤔

Reply:
Agreed with Pin, I will access the `CatalogReader` via `session.env().catalog_reader()` in the handler.